src/pages/en/guides/debugging.md

```diff
@@ -4,4 +4,64 @@ title: Debugging
 description: Debug in Astro using the Debug component
 ---
 
-Astro runs on the server and logs directly to your terminal, so it can be difficult to debug values from Astro. Astro’s built-in `<Debug>` component can help you inspect values inside your files on the clientside. Read more about the [built-in Debug Component](/en/reference/builtin-components#debug-).
+Astro provides several different tools to help you debug your code easier and faster.
+
+## Debugging with `console.log()`
+
+`console.log()` is a simple-but-popular method of debugging your Astro code, especially in the component front matter.  
+
+
+```astro
+---
+const sum = (a, b) => a + b;
+
+// Example: Outputs "4" to the CLI terminal
+console.log(sum(2, 2));
+---
+```
+
+A `console.log()` statement in Astro frontmatter will always output to the **terminal** running the Astro CLI. This is because Astro runs on the server, and never in the browser.
+
+## Debugging JS Scripts
+
+Code that is written or imported inside of an Astro `<script>` tag is run in the browser. Any `console.log()` statements or other debug otuput will be  printed to the console in your browser. 
+
+## Debugging Framework Components
+
+Framework components (like React and Svelte) are unique: They render server-side by default, meaning that console.log() debug output will be visible in the terminal. However, they can also be hydrated for the browser, which may cause your debug logs to also appear in the browser.
+
+This can be useful for debugging differences between the SSR output and the hydrated components in the browser.
+
+## Using the Astro Debug Component
+
+To help you debug your Astro components, Astro provides a built-in [`<Debug />`](/en/reference/builtin-components#debug-) component which renders any value directly into your component HTML template. This is useful for quick debugging in the browser without having to flip back-and-forth between your terminal and your browser.
+
+```astro
+---
+import { Debug } from 'astro/components';
+const sum = (a, b) => a + b;
+---
+
+<!-- Example: Outputs "{answer: 4}" to the browser -->
+<Debug answer={sum(2, 4)} />
+```
+
+The Debug component also supports object prop syntax for even more flexible and concise debugging:
+
+```astro
+---
+import { Debug } from 'astro/components';
+const sum = (a, b) => a + b;
+const answer = sum(2, 4);
+---
+<!-- Example: All three examples are equivilent. -->
+<Debug answer={sum(2, 4)} />
+<Debug {answer: sum(2, 4)} />
```

Comment:
This example needs double curly braces to work:

```suggestion
<Debug {{ answer: sum(2, 4) }} />
```

Reply:
Checked, and confirmed!